enclave_build/src/lib.rs

```diff
@@ -181,11 +184,23 @@ impl<'a> Docker2Eif<'a> {
             return Err(Docker2EifError::LinuxkitExecError);
         }
 
+        let arch = self.docker.architecture().map_err(|e| {
+            eprintln!("Docker error: {:?}", e);
+            Docker2EifError::DockerError
+        })?;
+
+        let flags = match arch.as_str() {
```

Comment:
I think the right abstraction is to pass the arch string to EifBuilder and let that object take the decision.
And this way you avoid making enclave_build dependent on eif_defs.

Reply:
This would require switching `eif_utils` dependent on `enclave_build/docker.rs`. Let's pass the flag instead of the arch string from here for now.